packages/nimble-components/src/table/template.ts

```diff
@@ -1,32 +1,22 @@
-import {
-    ElementsFilter,
-    html,
-    ref,
-    repeat,
-    slotted,
-    when
-} from '@microsoft/fast-element';
+import { children, html, ref, repeat, when } from '@microsoft/fast-element';
 import { DesignSystem } from '@microsoft/fast-foundation';
 import type { VirtualItem } from '@tanstack/virtual-core';
 import type { Table } from '.';
 import { TableHeader } from './components/header';
 import { TableRow } from './components/row';
-import { TableColumn } from '../table-column/base';
-
-const isTableColumn = (): ElementsFilter => {
-    const filter: ElementsFilter = (
-        value: Node,
-        _: number,
-        __: Node[]
-    ): boolean => {
-        return value instanceof TableColumn;
-    };
-    return filter;
-};
+import type { TableColumn } from '../table-column/base';
 
 // prettier-ignore
 export const template = html<Table>`
-    <template role="table">
+    <template
+        role="table"
+        ${children({
+        property: 'childItems',
+        attributeFilter: ['column-id'],
```

Comment:
I think it was alluded to last week when I shared the external observation stuff, but don't we ultimately want to be able to track all `observable` property changes to the column configuration, not just column-id?

Should we do the pattern in Meyer's PR to know when new observable objects exist: https://github.com/ni/nimble/blob/3ba7e2edc89e49028075247728c55cbc91e6bda6/packages/nimble-components/src/table/template.ts#L29

And register / unregister notifiers for them so we can tell when any `obserable` property changes?

Reply:
@atmgrifter00 and I debated this a bit, and we were on the fence about this approach vs registering/unregistering notifiers. The current implementation is less code because we're relying more on FAST, but I think you're right. We'll probably end up needing notifications about more things than the column-id, and we also might need more nuanced notification information regarding what changed than this approach can give us.

I'll switch this to manually registering & unregistering notifiers.